nwbwidgets/behavior.py

```diff
@@ -95,47 +72,134 @@ def show_spatial_series(node: SpatialSeries, **kwargs):
-            fig.update_xaxes(title_text="x ({})".format(unit))
-            fig.update_yaxes(title_text="y ({})".format(unit))
-        else:
-            fig.update_xaxes(title_text="x")
-            fig.update_yaxes(title_text="y")
-        fig.update_layout(height=600, width=600)
+    def set_out_fig(self):
+        timeseries = self.controls["timeseries"].value
+        time_window = self.controls["time_window"].value
+
+        istart = timeseries_time_to_ind(timeseries, time_window[0])
+        istop = timeseries_time_to_ind(timeseries, time_window[1])
+        data, units = get_timeseries_in_units(timeseries, istart, istop)
+
+        tt = get_timeseries_tt(timeseries, istart, istop)
+
+        self.plot_data(data, units, tt)
+
+        def on_change(change):
+            time_window = self.controls["time_window"].value
+            istart = timeseries_time_to_ind(timeseries, time_window[0])
+            istop = timeseries_time_to_ind(timeseries, time_window[1])
+            data, units = get_timeseries_in_units(timeseries, istart, istop)
+            tt = get_timeseries_tt(timeseries, istart, istop)
+            self.update_plot(data, tt)
+
+        self.controls["time_window"].observe(on_change)
+
+
+class SpatialSeriesTraceWidget2D(SpatialSeriesTraceWidget):
+    def plot_data(self, data, units, tt):
+        if units is None:
+            units = "no units"
+        self.out_fig = go.FigureWidget(
+            data=go.Scatter(
+                x=list(data[:, 0]),
+                y=list(data[:, 1]),
+                # mode="lines",
+                marker_color=tt,
+                marker_colorscale="Viridis",
+                marker_colorbar=dict(thickness=20, title="time (s)"),
+                marker_size=5,
+            )
+        )
 
-    elif data.shape[1] == 3:
-        fig.add_trace(go.Scatter3d(x=data[:, 0], y=data[:, 1], z=data[:, 2]))
+        self.out_fig.update_layout(
+            title=self.timeseries.name,
+            xaxis_title=f"x ({units})",
+            yaxis_title=f"y ({units})",
+        )
 
-        if unit:
-            fig.update_xaxes(title_text="x ({})".format(unit))
-            fig.update_yaxes(title_text="y ({})".format(unit))
-            # fig.update_zaxes(title_text='z ({})'.format(unit))
-        else:
-            fig.update_xaxes(title_text="x")
-            fig.update_yaxes(title_text="y")
-            # fig.update_zaxes(title_text='z')
+    def update_plot(self, data, tt):
+        self.out_fig.data[0].x = list(data[:, 0])
+        self.out_fig.data[0].y = list(data[:, 1])
+        self.out_fig.update_traces(marker_color=list(tt))
+
+
+class SpatialSeriesTraceWidget3D(SpatialSeriesTraceWidget):
+    def plot_data(self, data, units, tt):
+        if units is None:
+            units = "no units"
+
+        self.out_fig = go.FigureWidget(
+            data=go.Scatter3d(
+                x=data[:, 0],
+                y=data[:, 1],
+                z=data[:, 2],
+                # mode="lines",
```

Comment:
```suggestion
                mode="markers",
```
Being explicit about the mode makes the color scale appear on my browser. 
I think this is a great way to visually show 1) the speed of movement 2) places there can be discontinuity due to missing inter-trial data. Super!